src/torrent.cpp

```diff
@@ -1715,9 +1719,9 @@ namespace libtorrent
 			// is available
 			// copy the peer list since peers may disconnect and invalidate
 			// m_connections as we initialize them
-			std::vector<peer_connection*> peers = m_connections;
-			for (auto* pc :  peers)
+			for (auto c : clone(m_connections))
 			{
+				auto pc = c->self();
```

Comment:
this is necessary (I think) because `pc` is used after `pc->on_metadata_impl();`